homeassistant/components/sensoterra/sensor.py

```diff
@@ -0,0 +1,149 @@
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key=ProbeSensorType.LASTSEEN,
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        translation_key="last_seen",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SensoterraConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up Sensoterra sensor."""
+
+    coordinator = entry.runtime_data
+
+    @callback
+    def _async_add_devices(devices: dict[str, SensoterraSensor]) -> None:
+        async_add_devices(
+            SensoterraEntity(coordinator, sensor, description)
+            for sensor in devices.values()
+            for description in SENSORS
+            if sensor.type == description.key
+        )
+
+    coordinator.add_devices_callback = _async_add_devices
+
+    _async_add_devices(coordinator.data)
+
+
+class SensoterraEntity(CoordinatorEntity[SensoterraCoordinator], SensorEntity):
+    """Sensoterra sensor like a soil moisture or temperature sensor."""
+
+    def __init__(
+        self,
+        coordinator: SensoterraCoordinator,
+        sensor: SensoterraSensor,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(coordinator, context=sensor.id)
+
+        self._attr_unique_id = sensor.id
+        self._attr_has_entity_name = True
+
+        self.entity_description = description
+
+        # Make sure {depth} placeholders gets substituted.
+        self._attr_translation_placeholders = {"depth": str(sensor.depth)}
+        if isinstance(self.entity_description.name, str):
+            self._attr_name = self.entity_description.name.format(depth=sensor.depth)
+
+        # Add soil type to certain sensors.
+        if sensor.soil is not None and sensor.type in [
+            ProbeSensorType.MOISTURE,
+            ProbeSensorType.SI,
+        ]:
+            self._attr_extra_state_attributes = {"soil_type": sensor.soil}
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, sensor.serial)},
+            name=sensor.name,
+            model=sensor.sku,
+            manufacturer="Sensoterra",
+            serial_number=sensor.serial,
+            suggested_area=sensor.location,
+            configuration_url=CONFIGURATION_URL,
+        )
+
+        if sensor.value is not None:
+            self._attr_native_value = sensor.value
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        if self._attr_unique_id is not None:
+            sensor = self.coordinator.data[self._attr_unique_id]
+            if sensor.value is not None:
+                self._attr_native_value = sensor.value
+                self.async_write_ha_state()
```

Comment:
I'd just make a `native_value` property instead, that would make this simpler

Reply:
Could you please elaborate on this a little bit more? It's unclear to me what you suggest to consider.

Reply:
```py
@property
def native_value(self) -> StateType:
    return ....
```